packages/react-devtools-shared/src/devtools/store.js

```diff
@@ -1187,6 +1187,15 @@ export default class Store extends EventEmitter<{|
       console.groupEnd();
     }
 
+    const indicesOfCachedErrorsOrWarningsAreStale =
```

Comment:
I think this is also missing the case where we just re-order elements. We could let the operations decide when `indicesOfCachedErrorsOrWarningsAreStale` should be true instead but this feels a bit too brittle (though we have the same problem with `haveErrorsOrWarningsChanged`).

The other solution would be to rethink the index caching strategy alltogether. So just cache the IDs and get the indices (cached) when necessary.

Reply:
This is a good point.

Reply:
TBH we should probably just store the ID and look up the index on the fly each time (no caching).

Reply:
> TBH we should probably just store the ID and look up the index on the fly each time (no caching).

That was my thinking as well. Though it would need some additional work since we currently assume that `errorsOrWarnings` are sorted by indices.

So maybe we keeps an unsorted set with all the fiber ids with errors or warnings and a separate one that has them sorted by index. When we handle operations we just update the set and once the user wants to loop through them we create a (cached) list that has them ordered by index.